File: SEMANA I/introductorio_2017/c/gcd/gcd.c
#include <stdio.h>
#include <stdlib.h>

    // C plain
	int gcd(int a, int b){
	   if (a == 0) return b;
	   if (b == 0) return a;
		while ( a != b ){
			if ( a > b ) a = a - b;
			else b = b - a;	
		}
		return a;
	}



int  main (int argc, char *argv[]) {
	    int a = 0, b = 0;
		if ( argc >= 2  ){
			a = atoi(argv[1]);
		}
		if ( argc >= 3  ){
			b = atoi(argv[2]);
		}
		printf("gcd(%d, %d)= %d\n", a, b, gcd(a, b));
		return 0;
}

File: SEMANA I/introductorio_2017/python/gcd/gcd.py
import sys

def gcd(a, b) :
   if a == 0 : return b;
   if b == 0 : return a;
   while a != b :
     if a > b :
        a = a - b
     else :
        b = b - a
   return a

def main():
     a, b =  (0, 0)
     if len(sys.argv) >= 2 :
         a = int(sys.argv[1])
     if len(sys.argv)  >= 3 :
         b = int(sys.argv[2])
     print("gcd(%d, %d)= %d" % (a, b, gcd(a, b))) # Estilo 2.x
     print(f"gcd({a}, {b})= {gcd(a, b)}")         # Estilo 3.x
#     
main()

 